hudi-common/src/main/java/org/apache/hudi/common/table/HoodieTableConfig.java

```diff
@@ -632,6 +632,7 @@ static boolean validateConfigVersion(ConfigProperty<?> configProperty, HoodieTab
     // validate that the table version is greater than or equal to the config version
     HoodieTableVersion firstVersion = HoodieTableVersion.fromReleaseVersion(configProperty.getSinceVersion().get());
     boolean valid = tableVersion.greaterThan(firstVersion) || tableVersion.equals(firstVersion);
+    valid = valid || configProperty.key().equals(KEY_GENERATOR_CLASS_NAME.key()) || configProperty.key().equals(KEY_GENERATOR_TYPE.key());
```

Comment:
shouldn't this be &&?

Reply:
KEY_GENERATOR_CLASS_NAME  and KEY_GENERATOR_TYPE were added in table version 8 but are also used in table version 6 code. Valid flag fails for them when table version is 6 so need to check for these keys.

Reply:
lets discuss the right fix for this